{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5944fb7b",
   "metadata": {},
   "source": [
    "INITIALIZE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "87ae89f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import math "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "45cc74b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "mut_test = pd.read_excel(\"C:/Users/hheath/Downloads/mutation table chart.xlsx\")\n",
    "aakey = pd.read_excel(\"C:/Users/hheath/Documents/amino acid key.xlsx\")\n",
    "ube3a_seq = 'atggccacagcttgtaaaagatcaggagaacctcagtctgacgacattgaagctagccgaatgaagcgagcagctgcaaagcatctaatagaacgctactaccaccagttaactgagggctgtggaaatgaagcctgcacgaatgagttttgtgcttcctgtccaacttttcttcgtatggataataatgcagcagctattaaagccctcgagctttataagattaatgcaaaactctgtgatcctcatccctccaagaaaggagcaagctcagcttaccttgagaactcgaaaggtgcccccaacaactcctgctctgagataaaaatgaacaagaaaggcgctagaattgattttaaagatgtgacttacttaacagaagagaaggtatatgaaattcttgaattatgtagagaaagagaggattattcccctttaatccgtgttattggaagagttttttctagtgctgaggcattggtacagagcttccggaaagttaaacaacacaccaaggaagaactgaaatctcttcaagcaaaagatgaagacaaagatgaagatgaaaaggaaaaagctgcatgttctgctgctgctatggaagaagactcagaagcatcttcctcaaggataggtgatagctcacagggagacaacaatttgcaaaaattaggccctgatgatgtgtctgtggatattgatgccattagaagggtctacaccagattgctctctaatgaaaaaattgaaactgcctttctcaatgcacttgtatatttgtcacctaacgtggaatgtgacttgacgtatcacaatgtatactctcgagatcctaattatctgaatttgttcattatcgtaatggagaatagaaatctccacagtcctgaatatctggaaatggctttgccattattttgcaaagcgatgagcaagctaccccttgcagcccaaggaaaactgatcagactgtggtctaaatacaatgcagaccagattcggagaatgatggagacatttcagcaacttattacttataaagtcataagcaatgaatttaacagtcgaaatctagtgaatgatgatgatgccattgttgctgcttcgaagtgcttgaaaatggtttactatgcaaatgtagtgggaggggaagtggacacaaatcacaatgaagaagatgatgaagagcccatccctgagtccagcgagctgacacttcaggaacttttgggagaagaaagaagaaacaagaaaggtcctcgagtggaccccctggaaactgaacttggtgttaaaaccctggattgtcgaaaaccacttatcccttttgaagagtttattaatgaaccactgaatgaggttctagaaatggataaagattatacttttttcaaagtagaaacagagaacaaattctcttttatgacatgtccctttatattgaatgctgtcacaaagaatttgggattatattatgacaatagaattcgcatgtacagtgaacgaagaatcactgttctctacagcttagttcaaggacagcagttgaatccatatttgagactcaaagttagacgtgaccatatcatagatgatgcacttgtccggctagagatgatcgctatggaaaatcctgcagacttgaagaagcagttgtatgtggaatttgaaggagaacaaggagttgatgagggaggtgtttccaaagaattttttcagctggttgtggaggaaatcttcaatccagatattggtatgttcacatacgatgaatctacaaaattgttttggtttaatccatcttcttttgaaactgagggtcagtttactctgattggcatagtactgggtctggctatttacaataactgtatactggatgtacattttcccatggttgtctacaggaagctaatggggaaaaaaggaacttttcgtgacttgggagactctcacccagttctatatcagagtttaaaagatttattggagtatgaagggaatgtggaagatgacatgatgatcactttccagatatcacagacagatctttttggtaacccaatgatgtatgatctaaaggaaaatggtgataaaattccaattacaaatgaaaacaggaaggaatttgtcaatctttattctgactacattctcaataaatcagtagaaaaacagttcaaggcttttcggagaggttttcatatggtgaccaatgaatctcccttaaagtacttattcagaccagaagaaattgaattgcttatatgtggaagccggaatctagatttccaagcactagaagaaactacagaatatgacggtggctataccagggactctgttctgattagggagttctgggaaatcgttcattcatttacagatgaacagaaaagactcttcttgcagtttacaacgggcacagacagagcacctgtgggaggactaggaaaattaaagatgattatagccaaaaatggcccagacacagaaaggttacctacatctcatacttgctttaatgtgcttttacttccggaatactcaagcaaagaaaaacttaaagagagattgttgaaggccatcacgtatgccaaaggatttggcatgctgtaa'\n",
    "ube3a_seq = ube3a_seq.upper()\n",
    "aa = pd.read_excel(\"C:/Users/hheath/Documents/amino acids list_lei.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a68ecf55",
   "metadata": {},
   "outputs": [],
   "source": [
    "#split into codons, write into list\n",
    "ube3a_cod = []\n",
    "ube3a_codons = list(zip(*[iter(ube3a_seq)]*3))\n",
    "for i in range( 0, len(ube3a_codons)): \n",
    "    cod = ''.join(ube3a_codons[i])\n",
    "    cod = cod.upper()\n",
    "    ube3a_cod.append (cod)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1e9d4a78",
   "metadata": {},
   "source": [
    "PART 1: AMINO ACID ID, MUTATION LIST, AMINO ACID CHANGES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "89667d35",
   "metadata": {},
   "outputs": [],
   "source": [
    "#amino acid key\n",
    "ala = list(aakey.Ala)\n",
    "cys = list(aakey.Cys)\n",
    "asp = list(aakey.Asp)\n",
    "glu = list(aakey.Glu)\n",
    "phe = list(aakey.Phe)\n",
    "gly = list(aakey.Gly)\n",
    "his = list(aakey.His)\n",
    "ile = list(aakey.Ile)\n",
    "lys = list(aakey.Lys) \n",
    "leu = list(aakey.Leu)\n",
    "met = list(aakey.Met)\n",
    "pro = list(aakey.Pro)\n",
    "gln = list(aakey.Gln)\n",
    "arg = list(aakey.Arg)\n",
    "ser = list(aakey.Ser)\n",
    "thr = list(aakey.Thr)\n",
    "val = list(aakey.Val)\n",
    "trp = list(aakey.Trp)\n",
    "tyr = list(aakey.Tyr)\n",
    "stop= list(aakey.stop)\n",
    "asn = list(aakey.Asn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "3edbeeb0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def aa_id (codon): \n",
    "    if codon in ala: \n",
    "        return('Ala')\n",
    "    if codon in cys: \n",
    "        return('Cys')\n",
    "    if codon in asp: \n",
    "        return('Asp')\n",
    "    if codon in glu: \n",
    "        return('Glu')\n",
    "    if codon in phe: \n",
    "        return('Phe')\n",
    "    if codon in gly: \n",
    "        return('Gly')\n",
    "    if codon in his: \n",
    "        return('His')\n",
    "    if codon in ile: \n",
    "        return('Ile')\n",
    "    if codon in lys: \n",
    "        return('Lys')\n",
    "    if codon in leu: \n",
    "        return('Leu')\n",
    "    if codon in met: \n",
    "        return('Met')\n",
    "    if codon in pro: \n",
    "        return('Pro')\n",
    "    if codon in gln: \n",
    "        return('Gln')\n",
    "    if codon in arg: \n",
    "        return('Arg')\n",
    "    if codon in ser: \n",
    "        return('Ser')\n",
    "    if codon in thr: \n",
    "        return('Thr')\n",
    "    if codon in val: \n",
    "        return('Val')\n",
    "    if codon in trp: \n",
    "        return('Trp')\n",
    "    if codon in tyr: \n",
    "        return('Tyr')\n",
    "    if codon in stop: \n",
    "        return('STOP')  \n",
    "    if codon in asn: \n",
    "        return ('Asn')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "66b4f0e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def nuc1_change(codon): #returns 3 mutations of nucleotide 1, counts mutations resulting in unique amino acid changes\n",
    "    temp1 = list(codon)\n",
    "    temp2 = list(codon)\n",
    "    temp3 = list(codon)\n",
    "    temp4 = list(codon)\n",
    "    \n",
    "    mutlst = []                        #initialize mutation count\n",
    "    \n",
    "    if temp1[0] == 'C': \n",
    "        altc = 'original'\n",
    "        temp1 = ''.join(temp1)         #original codon designation\n",
    "    else:\n",
    "        temp1[0] = 'C'                 #change nucleotide #1 to C\n",
    "        temp1 = ''.join(temp1)\n",
    "        altc = aa_id(temp1)            #amino acid ID of mutation\n",
    "        if altc == aa_id(codon):       #AA change vs no change \n",
    "            altc = 'no_change'\n",
    "    \n",
    "    if temp2[0] == 'G': \n",
    "        altg = 'original'\n",
    "        temp2 = ''.join(temp2)\n",
    "    else:\n",
    "        temp2[0] = 'G'\n",
    "        temp2 = ''.join(temp2)\n",
    "        altg = aa_id(temp2)\n",
    "        if altg == aa_id(codon): \n",
    "            altg = 'no_change'     \n",
    "            \n",
    "    if temp3[0] == 'T': \n",
    "        altt = 'original'\n",
    "        temp3 = ''.join(temp3)\n",
    "    else:\n",
    "        temp3[0] = 'T'\n",
    "        temp3 = ''.join(temp3)\n",
    "        altt = aa_id(temp3)\n",
    "        if altt == aa_id(codon): \n",
    "            altt = 'no_change'\n",
    "            \n",
    "    if temp4[0] == 'A': \n",
    "        alta = 'original'\n",
    "        temp4 = ''.join(temp4)\n",
    "    else:\n",
    "        temp4[0] = 'A'\n",
    "        temp4 = ''.join(temp4)\n",
    "        alta = aa_id(temp4)\n",
    "        if alta == aa_id(codon): \n",
    "            alta = 'no_change'\n",
    "            \n",
    "    if altc != 'no_change' and altc != 'original':    #putting all mutations into a list\n",
    "        mutlst.append(altc)\n",
    "    if altg != 'no_change' and altg != 'original': \n",
    "        mutlst.append(altg)\n",
    "    if altt != 'no_change' and altt != 'original': \n",
    "        mutlst.append(altt)\n",
    "    if alta != 'no_change' and alta != 'original': \n",
    "        mutlst.append(alta)\n",
    "    mutset = set(mutlst)                              #change list to set to eliminate repeated values\n",
    "    mutations = len(mutset)                           #count items in the mutation set = number of unique outcomes\n",
    "    \n",
    "    if altc == 'original':                            #return original codon and aa, then alts, then mutation count\n",
    "        aaid1 = aa_id(temp1)\n",
    "        return temp1,aaid1,altc,temp2,altg,temp3,altt,temp4,alta,mutations\n",
    "    if altg == 'original': \n",
    "        aaid1 = aa_id(temp2)\n",
    "        return temp2,aaid1,altg,temp3,altt,temp4,alta,temp1,altc,mutations\n",
    "    if alta == 'original': \n",
    "        aaid1 = aa_id(temp4)\n",
    "        return temp4,aaid1,alta,temp1,altc,temp2,altg,temp3,altt,mutations\n",
    "    if altt == 'original': \n",
    "        aaid1 = aa_id(temp3)\n",
    "        return temp3,aaid1,altt,temp4,alta,temp1,altc,temp2,altg,mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "aba9a14d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def nuc2_change(codon):  #same code as nuc1_change, except operates on nucleotide 2\n",
    "    temp1 = list(codon)\n",
    "    temp2 = list(codon)\n",
    "    temp3 = list(codon)\n",
    "    temp4 = list(codon)\n",
    "    \n",
    "    mutlst = []\n",
    "    \n",
    "    if temp1[1] == 'C': \n",
    "        altc = 'original'\n",
    "        temp1 = ''.join(temp1)\n",
    "    else:\n",
    "        temp1[1] = 'C'\n",
    "        temp1 = ''.join(temp1)\n",
    "        altc = aa_id(temp1)\n",
    "        if altc == aa_id(codon): \n",
    "            altc = 'no_change'\n",
    "    \n",
    "    if temp2[1] == 'G': \n",
    "        altg = 'original'\n",
    "        temp2 = ''.join(temp2)\n",
    "    else:\n",
    "        temp2[1] = 'G'\n",
    "        temp2 = ''.join(temp2)\n",
    "        altg = aa_id(temp2)\n",
    "        if altg == aa_id(codon): \n",
    "            altg = 'no_change'       \n",
    "            \n",
    "    if temp3[1] == 'T': \n",
    "        altt = 'original'\n",
    "        temp3 = ''.join(temp3)\n",
    "    else:\n",
    "        temp3[1] = 'T'\n",
    "        temp3 = ''.join(temp3)\n",
    "        altt = aa_id(temp3)\n",
    "        if altt == aa_id(codon): \n",
    "            altt = 'no_change'\n",
    "            \n",
    "    if temp4[1] == 'A': \n",
    "        alta = 'original'\n",
    "        temp4 = ''.join(temp4)\n",
    "    else:\n",
    "        temp4[1] = 'A'\n",
    "        temp4 = ''.join(temp4)\n",
    "        alta = aa_id(temp4)\n",
    "        if alta == aa_id(codon): \n",
    "            alta = 'no_change'  \n",
    "    \n",
    "    if altc != 'no_change' and altc != 'original':    #mutation count\n",
    "        mutlst.append(altc)\n",
    "    if altg != 'no_change' and altg != 'original': \n",
    "        mutlst.append(altg)\n",
    "    if altt != 'no_change' and altt != 'original': \n",
    "        mutlst.append(altt)\n",
    "    if alta != 'no_change' and alta != 'original': \n",
    "        mutlst.append(alta)\n",
    "    mutset = set(mutlst)\n",
    "    mutations = len(mutset)\n",
    "            \n",
    "    if altc == 'original':\n",
    "        return temp1,altc,temp2,altg,temp3,altt,temp4,alta,mutations\n",
    "    if altg == 'original': \n",
    "        return temp2,altg,temp3,altt,temp4,alta,temp1,altc,mutations\n",
    "    if alta == 'original': \n",
    "        return temp4,alta,temp1,altc,temp2,altg,temp3,altt,mutations\n",
    "    if altt == 'original': \n",
    "        return temp3,altt,temp4,alta,temp1,altc,temp2,altg,mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "0f7104ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "def nuc3_change(codon): # same code as nuc1_change and nuc2_change; except operates on nucleotide 3\n",
    "    temp1 = list(codon)\n",
    "    temp2 = list(codon)\n",
    "    temp3 = list(codon)\n",
    "    temp4 = list(codon)\n",
    "    \n",
    "    mutlst = []\n",
    "    \n",
    "    if temp1[2] == 'C': \n",
    "        altc = 'original'\n",
    "        temp1 = ''.join(temp1)\n",
    "    else:\n",
    "        temp1[2] = 'C'\n",
    "        temp1 = ''.join(temp1)\n",
    "        altc = aa_id(temp1)\n",
    "        if altc == aa_id(codon): \n",
    "            altc = 'no_change'\n",
    "    \n",
    "    if temp2[2] == 'G': \n",
    "        altg = 'original'\n",
    "        temp2 = ''.join(temp2)\n",
    "    else:\n",
    "        temp2[2] = 'G'\n",
    "        temp2 = ''.join(temp2)\n",
    "        altg = aa_id(temp2)\n",
    "        if altg == aa_id(codon): \n",
    "            altg = 'no_change'\n",
    "            \n",
    "    if temp3[2] == 'T': \n",
    "        altt = 'original'\n",
    "        temp3 = ''.join(temp3)\n",
    "    else:\n",
    "        temp3[2] = 'T'\n",
    "        temp3 = ''.join(temp3)\n",
    "        altt = aa_id(temp3)\n",
    "        if altt == aa_id(codon): \n",
    "            altt = 'no_change'\n",
    "            \n",
    "    if temp4[2] == 'A': \n",
    "        alta = 'original'\n",
    "        temp4 = ''.join(temp4)\n",
    "    else:\n",
    "        temp4[2] = 'A'\n",
    "        temp4 = ''.join(temp4)\n",
    "        alta = aa_id(temp4)\n",
    "        if alta == aa_id(codon): \n",
    "            alta = 'no_change'       \n",
    "    \n",
    "    if altc != 'no_change' and altc != 'original':    #mutation count\n",
    "        mutlst.append(altc)\n",
    "    if altg != 'no_change' and altg != 'original': \n",
    "        mutlst.append(altg)\n",
    "    if altt != 'no_change' and altt != 'original': \n",
    "        mutlst.append(altt)\n",
    "    if alta != 'no_change' and alta != 'original': \n",
    "        mutlst.append(alta)\n",
    "    mutset = set(mutlst)\n",
    "    mutations = len(mutset)\n",
    "            \n",
    "    if altc == 'original': \n",
    "        return temp1,altc,temp2,altg,temp3,altt,temp4,alta,mutations\n",
    "    if altg == 'original': \n",
    "        return temp2,altg,temp3,altt,temp4,alta,temp1,altc,mutations\n",
    "    if alta == 'original': \n",
    "        return temp4,alta,temp1,altc,temp2,altg,temp3,altt,mutations\n",
    "    if altt == 'original': \n",
    "        return temp3,altt,temp4,alta,temp1,altc,temp2,altg,mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "c163f79b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#test list\n",
    "test = ube3a_cod[0:21]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "80b6b0d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#nuc1 ref list for testing\n",
    "aaid_list = []\n",
    "for i in range (0, len(test)): \n",
    "    aaid = aa_id(test[i])\n",
    "    aaid_list.append(aaid)\n",
    "ref = pd.DataFrame(aaid_list, test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "f0f5c165",
   "metadata": {},
   "outputs": [],
   "source": [
    "#codon number starting at 1\n",
    "codnum = np.arange(1, (len(ube3a_cod)+1))\n",
    "codnumdf = pd.DataFrame(codnum, columns = ['codon #'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "51f8d884",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create dataframes for nuc1, nuc2, and nuc3\n",
    "n1res = []\n",
    "for i in range(0,len(ube3a_cod)): \n",
    "    n1res.append(nuc1_change(ube3a_cod[i]))\n",
    "res1 = pd.DataFrame(n1res)\n",
    "\n",
    "n2res = []\n",
    "for i in range(0,len(ube3a_cod)): \n",
    "    n2res.append(nuc2_change(ube3a_cod[i]))\n",
    "res2 = pd.DataFrame(n2res)\n",
    "\n",
    "n3res = []\n",
    "for i in range(0,len(ube3a_cod)): \n",
    "    n3res.append(nuc3_change(ube3a_cod[i]))\n",
    "res3 = pd.DataFrame(n3res)\n",
    "\n",
    "res1.columns = ['original codon', 'AA ID', 'ignore', 'nuc1 alt1', 'aa1 alt1', 'nuc1 alt2', 'aa1 alt2', 'nuc1 alt3', 'aa1 alt3', 'mutations']\n",
    "res2.columns = ['original codon', 'ignore', 'nuc2 alt1', 'aa2 alt1', 'nuc2 alt2', 'aa2 alt2', 'nuc2 alt3', 'aa2 alt3', 'mutations']\n",
    "res3.columns = ['original codon', 'ignore', 'nuc3 alt1', 'aa3 alt1', 'nuc3 alt2', 'aa3 alt2', 'nuc3 alt3', 'aa3 alt3', 'mutations']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "1d9ed77a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#combines codon #, nuc1, nuc2, and nuc3 into one dataframe (add columns, not rows)\n",
    "#writes to excel file 'aminotest2'\n",
    "aminochange = pd.concat([codnumdf,res1,res2,res3], axis = 1)\n",
    "with pd.ExcelWriter(\"C:/Users/hheath/Documents/aminotest2.xlsx\") as writer:\n",
    "    aminochange.to_excel(writer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "25e99f6f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#UN-SET, DELETE NO_CHANGE, THEN USE SET() TO GET RID OF REPEATS, CONCAT INTO A DF\n",
    "muts = []\n",
    "for i in range (0, len(ube3a_cod)): \n",
    "    muts.append(flip(res1['aa1 alt1'], res1['aa1 alt2'], res1['aa1 alt3'], res2['aa2 alt1'],res2['aa2 alt2'],res2['aa2 alt3'],res3['aa3 alt1'],res3['aa3 alt2'],res3['aa3 alt3'],i))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "18621a53",
   "metadata": {},
   "outputs": [],
   "source": [
    "def mutlst (genome, ind): \n",
    "    NCcount = 0\n",
    "    STcount = 0 \n",
    "    lst = genome[ind]\n",
    "    for i in range(0, len(lst)): \n",
    "        if lst[i] == 'no_change': \n",
    "            NCcount += 1\n",
    "        if lst[i] == 'STOP': \n",
    "            STcount += 1\n",
    "    for i in range(0, NCcount): \n",
    "        lst.remove('no_change')\n",
    "    for i in range(0, STcount): \n",
    "        lst.remove('STOP')\n",
    "    lst = set(lst)\n",
    "    return lst"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e62ab66-722f-4f2c-94e1-a58692d5d938",
   "metadata": {},
   "source": [
    "SIDE QUEST TO FILL OUT LEI'S FORM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "926a4ea7",
   "metadata": {},
   "outputs": [],
   "source": [
    "formlist = []\n",
    "for i in range(0, len(muts)): \n",
    "    formlist.append(mutlst(muts, i))\n",
    "frmlst = pd.DataFrame(formlist)\n",
    "with pd.ExcelWriter(\"C:/Users/hheath/Documents/aminotest2.xlsx\") as writer:\n",
    "    frmlst.to_excel(writer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "aa75b0f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "aalist1 = ['A','C','D','E','F','G','H','I','K','L','M','N','P','Q','R','S','T','V','W','Y']\n",
    "formmut = pd.read_excel('C:/Users/hheath/Documents/mutationsform.xlsx')\n",
    "formmut.columns\n",
    "cod = list(formmut['codnum'])\n",
    "mut1 = list(formmut['mut1'])\n",
    "mut2 = list(formmut['mut2'])\n",
    "mut3 = list(formmut['mut3'])\n",
    "mut4 = list(formmut['mut4'])\n",
    "mut5 = list(formmut['mut5'])\n",
    "mut6 = list(formmut['mut6'])\n",
    "mut7 = list(formmut['mut7'])\n",
    "dfmut = pd.DataFrame(mut1, cod)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "530b2231",
   "metadata": {},
   "outputs": [],
   "source": [
    "def AAsort(amutlst, amutlst2, amutlst3, amutlst4, amutlst5, amutlst6, amutlst7):\n",
    "    A = []\n",
    "    C = []\n",
    "    D = []\n",
    "    E = []\n",
    "    F = []\n",
    "    G = []\n",
    "    H = []\n",
    "    I = []\n",
    "    K = []\n",
    "    L = []\n",
    "    M = []\n",
    "    N = []\n",
    "    P = []\n",
    "    Q = []\n",
    "    R = []\n",
    "    S = []\n",
    "    T = []\n",
    "    V = []\n",
    "    W = []\n",
    "    Y = []\n",
    "    for i in range (0, len(amutlst)):\n",
    "        if amutlst[i] == 'A' or amutlst2[i] == 'A' or amutlst3[i] == 'A' or amutlst4[i] == 'A' or amutlst5[i] == 'A' or amutlst6[i] == 'A' or amutlst7[i] == 'A' : \n",
    "            A.append(ube3a_cod[i])    #REPLACE UBE3A_COD WITH THE MUTATIONS\n",
    "        else: \n",
    "            A.append('')\n",
    "        \n",
    "        if amutlst[i] == 'C' or amutlst2[i] == 'C' or amutlst3[i] == 'C' or amutlst4[i] == 'C' or amutlst5[i] == 'C' or amutlst6[i] == 'C' or amutlst7[i] == 'C' :\n",
    "            C.append(ube3a_cod[i])\n",
    "        else: \n",
    "            C.append('')\n",
    "        \n",
    "        if amutlst[i] == 'D' or amutlst2[i] == 'D' or amutlst3[i] == 'D' or amutlst4[i] == 'D' or amutlst5[i] == 'D' or amutlst6[i] == 'D' or amutlst7[i] == 'D' :\n",
    "            D.append(ube3a_cod[i])\n",
    "        else: \n",
    "            D.append('')\n",
    "            \n",
    "        if amutlst[i] == 'E' or amutlst2[i] == 'E' or amutlst3[i] == 'E' or amutlst4[i] == 'E' or amutlst5[i] == 'E' or amutlst6[i] == 'E' or amutlst7[i] == 'E' : \n",
    "            E.append(ube3a_cod[i])\n",
    "        else: \n",
    "            E.append('')\n",
    "        \n",
    "        if amutlst[i] == 'F' or amutlst2[i] == 'F' or amutlst3[i] == 'F' or amutlst4[i] == 'F' or amutlst5[i] == 'F' or amutlst6[i] == 'F' or amutlst7[i] == 'F' : \n",
    "            F.append(ube3a_cod[i])\n",
    "        else: \n",
    "            F.append('')\n",
    "    \n",
    "        if amutlst[i] == 'G' or amutlst2[i] == 'G' or amutlst3[i] == 'G' or amutlst4[i] == 'G' or amutlst5[i] == 'G' or amutlst6[i] == 'G' or amutlst7[i] == 'G' :\n",
    "            G.append(ube3a_cod[i])\n",
    "        else: \n",
    "            G.append('')\n",
    "        \n",
    "        if amutlst[i] == 'H' or amutlst2[i] == 'H' or amutlst3[i] == 'H' or amutlst4[i] == 'H' or amutlst5[i] == 'H' or amutlst6[i] == 'H' or amutlst7[i] == 'H' : \n",
    "            H.append(ube3a_cod[i])\n",
    "        else: \n",
    "            H.append('')\n",
    "        \n",
    "        if amutlst[i] == 'I' or amutlst2[i] == 'I' or amutlst3[i] == 'I' or amutlst4[i] == 'I' or amutlst5[i] == 'I' or amutlst6[i] == 'I' or amutlst7[i] == 'I' : \n",
    "            I.append(ube3a_cod[i])\n",
    "        else: \n",
    "            I.append('')\n",
    "        \n",
    "        if amutlst[i] == 'K' or amutlst2[i] == 'K' or amutlst3[i] == 'K' or amutlst4[i] == 'K' or amutlst5[i] == 'K' or amutlst6[i] == 'K' or amutlst7[i] == 'K' : \n",
    "            K.append(ube3a_cod[i])\n",
    "        else: \n",
    "            K.append('')\n",
    "    \n",
    "        if amutlst[i] == 'L' or amutlst2[i] == 'L' or amutlst3[i] == 'L' or amutlst4[i] == 'L' or amutlst5[i] == 'L' or amutlst6[i] == 'L' or amutlst7[i] == 'L' : \n",
    "            L.append(i+1)\n",
    "        else: \n",
    "            L.append('')\n",
    "        \n",
    "        if amutlst[i] == 'M' or amutlst2[i] == 'M' or amutlst3[i] == 'M' or amutlst4[i] == 'M' or amutlst5[i] == 'M' or amutlst6[i] == 'M' or amutlst7[i] == 'M' : \n",
    "            M.append(ube3a_cod[i])\n",
    "        else: \n",
    "            M.append('')\n",
    "        \n",
    "        if amutlst[i] == 'N' or amutlst2[i] == 'N' or amutlst3[i] == 'N' or amutlst4[i] == 'N' or amutlst5[i] == 'N' or amutlst6[i] == 'N' or amutlst7[i] == 'N' : \n",
    "            N.append(ube3a_cod[i])\n",
    "        else: \n",
    "            N.append('') \n",
    "        \n",
    "        if amutlst[i] == 'P' or amutlst2[i] == 'P' or amutlst3[i] == 'P' or amutlst4[i] == 'P' or amutlst5[i] == 'P' or amutlst6[i] == 'P' or amutlst7[i] == 'P' : \n",
    "            P.append(ube3a_cod[i])\n",
    "        else: \n",
    "            P.append('')\n",
    "    \n",
    "        if amutlst[i] == 'Q' or amutlst2[i] == 'Q' or amutlst3[i] == 'Q' or amutlst4[i] == 'Q' or amutlst5[i] == 'Q' or amutlst6[i] == 'Q' or amutlst7[i] == 'Q' : \n",
    "            Q.append(ube3a_cod[i])\n",
    "        else: \n",
    "            Q.append('')\n",
    "        \n",
    "        if amutlst[i] == 'R' or amutlst2[i] == 'R' or amutlst3[i] == 'R' or amutlst4[i] == 'R' or amutlst5[i] == 'R' or amutlst6[i] == 'R' or amutlst7[i] == 'R' : \n",
    "            R.append(ube3a_cod[i])\n",
    "        else: \n",
    "            R.append('')\n",
    "            \n",
    "        if amutlst[i] == 'S' or amutlst2[i] == 'S' or amutlst3[i] == 'S' or amutlst4[i] == 'S' or amutlst5[i] == 'S' or amutlst6[i] == 'S' or amutlst7[i] == 'S' : \n",
    "            S.append(ube3a_cod[i])\n",
    "        else: \n",
    "            S.append('') \n",
    "        \n",
    "        if amutlst[i] == 'T' or amutlst2[i] == 'T' or amutlst3[i] == 'T' or amutlst4[i] == 'T' or amutlst5[i] == 'T' or amutlst6[i] == 'T' or amutlst7[i] == 'T' : \n",
    "            T.append(ube3a_cod[i])\n",
    "        else: \n",
    "            T.append('')\n",
    "    \n",
    "        if amutlst[i] == 'V' or amutlst2[i] == 'V' or amutlst3[i] == 'V' or amutlst4[i] == 'V' or amutlst5[i] == 'V' or amutlst6[i] == 'V' or amutlst7[i] == 'V' : \n",
    "            V.append(ube3a_cod[i])\n",
    "        else: \n",
    "            V.append('')\n",
    "        \n",
    "        if amutlst[i] == 'W' or amutlst2[i] == 'W' or amutlst3[i] == 'W' or amutlst4[i] == 'W' or amutlst5[i] == 'W' or amutlst6[i] == 'W' or amutlst7[i] == 'W' : \n",
    "            W.append(ube3a_cod[i])\n",
    "        else: \n",
    "            W.append('')\n",
    "        \n",
    "        if amutlst[i] == 'Y' or amutlst2[i] == 'Y' or amutlst3[i] == 'Y' or amutlst4[i] == 'Y' or amutlst5[i] == 'Y' or amutlst6[i] == 'Y' or amutlst7[i] == 'Y' : \n",
    "            Y.append(ube3a_cod[i])\n",
    "        else: \n",
    "            Y.append('')\n",
    "    \n",
    "    A = pd.DataFrame(A)\n",
    "    C = pd.DataFrame(C)     \n",
    "    D = pd.DataFrame(D)\n",
    "    E = pd.DataFrame(E)\n",
    "    F = pd.DataFrame(F)\n",
    "    G = pd.DataFrame(G)\n",
    "    H = pd.DataFrame(H)\n",
    "    I = pd.DataFrame(I)       \n",
    "    K = pd.DataFrame(K)\n",
    "    L = pd.DataFrame(L)\n",
    "    M = pd.DataFrame(M)\n",
    "    N = pd.DataFrame(N)\n",
    "    P = pd.DataFrame(P)\n",
    "    Q = pd.DataFrame(Q)\n",
    "    R = pd.DataFrame(R)\n",
    "    S = pd.DataFrame(S)\n",
    "    T = pd.DataFrame(T)\n",
    "    V = pd.DataFrame(V)\n",
    "    W = pd.DataFrame(W)\n",
    "    Y = pd.DataFrame(Y)\n",
    "    \n",
    "    df = pd.concat([A,C,D,E,F,G,H,I,K,L,M,N,P,Q,R,S,T,V,W,Y], axis = 1)\n",
    "    df.columns = aalist1\n",
    "\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "673f5508",
   "metadata": {},
   "outputs": [],
   "source": [
    "#FILLS OUT LEI'S FORM\n",
    "cods =pd.DataFrame(np.arange(1, len(ube3a_cod)+1))\n",
    "XLmuttemp = AAsort(mut1, mut2,mut3,mut4,mut5,mut6,mut7)\n",
    "or_cod = pd.DataFrame(ube3a_cod)\n",
    "XLmut = pd.concat([cods, or_cod,XLmuttemp], axis = 1)\n",
    "with pd.ExcelWriter(\"C:/Users/hheath/Documents/aminotest.xlsx\") as writer:\n",
    "    XLmut.to_excel(writer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "d68880b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(('TCA', 'Ser', 'original', 'ACA', 'Thr', 'CCA', 'Pro', 'GCA', 'Ala', 3), ('TCA', 'original', 'TGA', 'STOP', 'TTA', 'Leu', 'TAA', 'STOP', 2), ('TCA', 'original', 'TCC', 'no_change', 'TCG', 'no_change', 'TCT', 'no_change', 0))\n",
      "TTA\n"
     ]
    }
   ],
   "source": [
    "nucmuts = nuc1_change(ube3a_cod[7]), nuc2_change(ube3a_cod[7]), nuc3_change(ube3a_cod[7])\n",
    "print(nucmuts)\n",
    "for i in range(0, 3): \n",
    "    temp = nucmuts[i] \n",
    "    if temp[4] == 'Leu': \n",
    "        print(temp[3])\n",
    "    if temp[6] == 'Leu': \n",
    "        print(temp[5])\n",
    "    if temp[8] == 'Leu': \n",
    "        print(temp[7])\n",
    "    if temp[3] == 'Leu': \n",
    "        print(temp[2])\n",
    "    if temp[5] == 'Leu': \n",
    "        print(temp[4])\n",
    "    if temp[7] == 'Leu': \n",
    "        print(temp[6])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "de16227b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def formfill_nuc(genome, ind): \n",
    "    A = []\n",
    "    C = []\n",
    "    D = []\n",
    "    E = []\n",
    "    F = []\n",
    "    G = []\n",
    "    H = []\n",
    "    I = []\n",
    "    K = []\n",
    "    L = []\n",
    "    M = []\n",
    "    N = []\n",
    "    P = []\n",
    "    Q = []\n",
    "    R = []\n",
    "    S = []\n",
    "    T = []\n",
    "    V = []\n",
    "    W = []\n",
    "    Y = []\n",
    "    \n",
    "    \n",
    "    codL = []\n",
    "    nucmuts = nuc1_change(genome[ind]), nuc2_change(genome[ind]), nuc3_change(genome[ind])\n",
    "    \n",
    "    Atemp = [] #will append to A\n",
    "    Ctemp = []\n",
    "    Dtemp = []\n",
    "    Etemp = []\n",
    "    Ftemp = []\n",
    "    Gtemp = []\n",
    "    Htemp = []\n",
    "    Itemp = []\n",
    "    Ktemp = []\n",
    "    Ltemp = []\n",
    "    Mtemp = []\n",
    "    Ntemp = []\n",
    "    Ptemp = []\n",
    "    Qtemp = []\n",
    "    Rtemp = []\n",
    "    Stemp = []\n",
    "    Ttemp = []\n",
    "    Vtemp = []\n",
    "    Wtemp = []\n",
    "    Ytemp = []\n",
    "    \n",
    "    \n",
    "    for j in range(0,3): \n",
    "        temp = nucmuts[j]\n",
    "        \n",
    "        if temp[4] == 'Ala':\n",
    "            Atemp.append(temp[3])\n",
    "        if temp[6] == 'Ala': \n",
    "            Atemp.append(temp[5])\n",
    "        if temp[8] == 'Ala': \n",
    "            Atemp.append(temp[7])\n",
    "        if temp[3] == 'Ala': \n",
    "            Atemp.append(temp[2])\n",
    "        if temp[5] == 'Ala': \n",
    "            Atemp.append(temp[4])\n",
    "        if temp[7] == 'Ala': \n",
    "            Atemp.append(temp[6])\n",
    "            \n",
    "        if temp[4] == 'Cys':\n",
    "            Ctemp.append(temp[3])\n",
    "        if temp[6] == 'Cys': \n",
    "            Ctemp.append(temp[5])\n",
    "        if temp[8] == 'Cys': \n",
    "            Ctemp.append(temp[7])\n",
    "        if temp[3] == 'Cys': \n",
    "            Ctemp.append(temp[2])\n",
    "        if temp[5] == 'Cys': \n",
    "            Ctemp.append(temp[4])\n",
    "        if temp[7] == 'Cys': \n",
    "            Ctemp.append(temp[6])\n",
    "        \n",
    "        if temp[4] == 'Asp':\n",
    "            Dtemp.append(temp[3])\n",
    "        if temp[6] == 'Asp': \n",
    "            Dtemp.append(temp[5])\n",
    "        if temp[8] == 'Asp': \n",
    "            Dtemp.append(temp[7])\n",
    "        if temp[3] == 'Asp': \n",
    "            Dtemp.append(temp[2])\n",
    "        if temp[5] == 'Asp': \n",
    "            Dtemp.append(temp[4])\n",
    "        if temp[7] == 'Asp': \n",
    "            Dtemp.append(temp[6])\n",
    "            \n",
    "        if temp[4] == 'Glu':\n",
    "            Etemp.append(temp[3])\n",
    "        if temp[6] == 'Glu': \n",
    "            Etemp.append(temp[5])\n",
    "        if temp[8] == 'Glu': \n",
    "            Etemp.append(temp[7])\n",
    "        if temp[3] == 'Glu': \n",
    "            Etemp.append(temp[2])\n",
    "        if temp[5] == 'Glu': \n",
    "            Etemp.append(temp[4])\n",
    "        if temp[7] == 'Glu': \n",
    "            Etemp.append(temp[6])\n",
    "            \n",
    "        if temp[4] == 'Phe':\n",
    "            Ftemp.append(temp[3])\n",
    "        if temp[6] == 'Phe': \n",
    "            Ftemp.append(temp[5])\n",
    "        if temp[8] == 'Phe': \n",
    "            Ftemp.append(temp[7])\n",
    "        if temp[3] == 'Phe': \n",
    "            Ftemp.append(temp[2])\n",
    "        if temp[5] == 'Phe': \n",
    "            Ftemp.append(temp[4])\n",
    "        if temp[7] == 'Phe': \n",
    "            Ftemp.append(temp[6])\n",
    "        \n",
    "        if temp[4] == 'Gly':\n",
    "            Gtemp.append(temp[3])\n",
    "        if temp[6] == 'Gly': \n",
    "            Gtemp.append(temp[5])\n",
    "        if temp[8] == 'Gly': \n",
    "            Gtemp.append(temp[7])\n",
    "        if temp[3] == 'Gly': \n",
    "            Gtemp.append(temp[2])\n",
    "        if temp[5] == 'Gly': \n",
    "            Gtemp.append(temp[4])\n",
    "        if temp[7] == 'Gly': \n",
    "            Gtemp.append(temp[6])\n",
    "        \n",
    "        if temp[4] == 'His':\n",
    "            Htemp.append(temp[3])\n",
    "        if temp[6] == 'His': \n",
    "            Htemp.append(temp[5])\n",
    "        if temp[8] == 'His': \n",
    "            Htemp.append(temp[7])\n",
    "        if temp[3] == 'His': \n",
    "            Htemp.append(temp[2])\n",
    "        if temp[5] == 'His': \n",
    "            Htemp.append(temp[4])\n",
    "        if temp[7] == 'His': \n",
    "            Htemp.append(temp[6])\n",
    "        \n",
    "        if temp[4] == 'Ile':\n",
    "            Itemp.append(temp[3])\n",
    "        if temp[6] == 'Ile': \n",
    "            Itemp.append(temp[5])\n",
    "        if temp[8] == 'Ile': \n",
    "            Itemp.append(temp[7])\n",
    "        if temp[3] == 'Ile': \n",
    "            Itemp.append(temp[2])\n",
    "        if temp[5] == 'Ile': \n",
    "            Itemp.append(temp[4])\n",
    "        if temp[7] == 'Ile': \n",
    "            Itemp.append(temp[6])\n",
    "            \n",
    "        if temp[4] == 'Lys':\n",
    "            Ktemp.append(temp[3])\n",
    "        if temp[6] == 'Lys': \n",
    "            Ktemp.append(temp[5])\n",
    "        if temp[8] == 'Lys': \n",
    "            Ktemp.append(temp[7])\n",
    "        if temp[3] == 'Lys': \n",
    "            Ktemp.append(temp[2])\n",
    "        if temp[5] == 'Lys': \n",
    "            Ktemp.append(temp[4])\n",
    "        if temp[7] == 'Lys': \n",
    "            Ktemp.append(temp[6])\n",
    "        \n",
    "        if temp[4] == 'Leu':\n",
    "            Ltemp.append(temp[3])\n",
    "        if temp[6] == 'Leu': \n",
    "            Ltemp.append(temp[5])\n",
    "        if temp[8] == 'Leu': \n",
    "            Ltemp.append(temp[7])\n",
    "        if temp[3] == 'Leu': \n",
    "            Ltemp.append(temp[2])\n",
    "        if temp[5] == 'Leu': \n",
    "            Ltemp.append(temp[4])\n",
    "        if temp[7] == 'Leu': \n",
    "            Ltemp.append(temp[6])\n",
    "            \n",
    "        if temp[4] == 'Met':\n",
    "            Mtemp.append(temp[3])\n",
    "        if temp[6] == 'Met': \n",
    "            Mtemp.append(temp[5])\n",
    "        if temp[8] == 'Met': \n",
    "            Mtemp.append(temp[7])\n",
    "        if temp[3] == 'Met': \n",
    "            Mtemp.append(temp[2])\n",
    "        if temp[5] == 'Met': \n",
    "            Mtemp.append(temp[4])\n",
    "        if temp[7] == 'Met': \n",
    "            Mtemp.append(temp[6])\n",
    "        \n",
    "        if temp[4] == 'Asn':\n",
    "            Ntemp.append(temp[3])\n",
    "        if temp[6] == 'Asn': \n",
    "            Ntemp.append(temp[5])\n",
    "        if temp[8] == 'Asn': \n",
    "            Ntemp.append(temp[7])\n",
    "        if temp[3] == 'Asn': \n",
    "            Ntemp.append(temp[2])\n",
    "        if temp[5] == 'Asn': \n",
    "            Ntemp.append(temp[4])\n",
    "        if temp[7] == 'Asn': \n",
    "            Ntemp.append(temp[6])\n",
    "        \n",
    "        if temp[4] == 'Pro':\n",
    "            Ptemp.append(temp[3])\n",
    "        if temp[6] == 'Pro': \n",
    "            Ptemp.append(temp[5])\n",
    "        if temp[8] == 'Pro': \n",
    "            Ptemp.append(temp[7])\n",
    "        if temp[3] == 'Pro': \n",
    "            Ptemp.append(temp[2])\n",
    "        if temp[5] == 'Pro': \n",
    "            Ptemp.append(temp[4])\n",
    "        if temp[7] == 'Pro': \n",
    "            Ptemp.append(temp[6])\n",
    "        \n",
    "        if temp[4] == 'Gln':\n",
    "            Qtemp.append(temp[3])\n",
    "        if temp[6] == 'Gln': \n",
    "            Qtemp.append(temp[5])\n",
    "        if temp[8] == 'Gln': \n",
    "            Qtemp.append(temp[7])\n",
    "        if temp[3] == 'Gln': \n",
    "            Qtemp.append(temp[2])\n",
    "        if temp[5] == 'Gln': \n",
    "            Qtemp.append(temp[4])\n",
    "        if temp[7] == 'Gln': \n",
    "            Qtemp.append(temp[6])\n",
    "        \n",
    "        if temp[4] == 'Arg':\n",
    "            Rtemp.append(temp[3])\n",
    "        if temp[6] == 'Arg': \n",
    "            Rtemp.append(temp[5])\n",
    "        if temp[8] == 'Arg': \n",
    "            Rtemp.append(temp[7])\n",
    "        if temp[3] == 'Arg': \n",
    "            Rtemp.append(temp[2])\n",
    "        if temp[5] == 'Arg': \n",
    "            Rtemp.append(temp[4])\n",
    "        if temp[7] == 'Arg': \n",
    "            Rtemp.append(temp[6])\n",
    "            \n",
    "        if temp[4] == 'Ser':\n",
    "            Stemp.append(temp[3])\n",
    "        if temp[6] == 'Ser': \n",
    "            Stemp.append(temp[5])\n",
    "        if temp[8] == 'Ser': \n",
    "            Stemp.append(temp[7])\n",
    "        if temp[3] == 'Ser': \n",
    "            Stemp.append(temp[2])\n",
    "        if temp[5] == 'Ser': \n",
    "            Stemp.append(temp[4])\n",
    "        if temp[7] == 'Ser': \n",
    "            Stemp.append(temp[6])\n",
    "            \n",
    "        if temp[4] == 'Thr':\n",
    "            Ttemp.append(temp[3])\n",
    "        if temp[6] == 'Thr': \n",
    "            Ttemp.append(temp[5])\n",
    "        if temp[8] == 'Thr': \n",
    "            Ttemp.append(temp[7])\n",
    "        if temp[3] == 'Thr': \n",
    "            Ttemp.append(temp[2])\n",
    "        if temp[5] == 'Thr': \n",
    "            Ttemp.append(temp[4])\n",
    "        if temp[7] == 'Thr': \n",
    "            Ttemp.append(temp[6])\n",
    "            \n",
    "        if temp[4] == 'Val':\n",
    "            Vtemp.append(temp[3])\n",
    "        if temp[6] == 'Val': \n",
    "            Vtemp.append(temp[5])\n",
    "        if temp[8] == 'Val': \n",
    "            Vtemp.append(temp[7])\n",
    "        if temp[3] == 'Val': \n",
    "            Vtemp.append(temp[2])\n",
    "        if temp[5] == 'Val': \n",
    "            Vtemp.append(temp[4])\n",
    "        if temp[7] == 'Val': \n",
    "            Vtemp.append(temp[6])\n",
    "            \n",
    "        if temp[4] == 'Trp':\n",
    "            Wtemp.append(temp[3])\n",
    "        if temp[6] == 'Trp': \n",
    "            Wtemp.append(temp[5])\n",
    "        if temp[8] == 'Trp': \n",
    "            Wtemp.append(temp[7])\n",
    "        if temp[3] == 'Trp': \n",
    "            Wtemp.append(temp[2])\n",
    "        if temp[5] == 'Trp': \n",
    "            Wtemp.append(temp[4])\n",
    "        if temp[7] == 'Trp': \n",
    "            Wtemp.append(temp[6])\n",
    "            \n",
    "        if temp[4] == 'Tyr':\n",
    "            Ytemp.append(temp[3])\n",
    "        if temp[6] == 'Tyr': \n",
    "            Ytemp.append(temp[5])\n",
    "        if temp[8] == 'Tyr': \n",
    "            Ytemp.append(temp[7])\n",
    "        if temp[3] == 'Tyr': \n",
    "            Ytemp.append(temp[2])\n",
    "        if temp[5] == 'Tyr': \n",
    "            Ytemp.append(temp[4])\n",
    "        if temp[7] == 'Tyr': \n",
    "            Ytemp.append(temp[6])\n",
    "            \n",
    "    A.append(Atemp)\n",
    "    C.append(Ctemp)\n",
    "    D.append(Dtemp)\n",
    "    E.append(Etemp)\n",
    "    F.append(Ftemp)\n",
    "    G.append(Gtemp)\n",
    "    H.append(Htemp)\n",
    "    I.append(Itemp)\n",
    "    K.append(Ktemp)\n",
    "    L.append(Ltemp)\n",
    "    M.append(Mtemp)\n",
    "    N.append(Ntemp)\n",
    "    P.append(Ptemp)\n",
    "    Q.append(Qtemp)\n",
    "    R.append(Rtemp)\n",
    "    S.append(Stemp)\n",
    "    T.append(Ttemp)\n",
    "    V.append(Vtemp)\n",
    "    W.append(Wtemp)\n",
    "    Y.append(Ytemp)\n",
    "    '''\n",
    "    A = pd.DataFrame(A)\n",
    "    C = pd.DataFrame(C)\n",
    "    D = pd.DataFrame(D)\n",
    "    E = pd.DataFrame(E)\n",
    "    F = pd.DataFrame(F)\n",
    "    G = pd.DataFrame(G)\n",
    "    H = pd.DataFrame(H)\n",
    "    I = pd.DataFrame(I)\n",
    "    K = pd.DataFrame(K)\n",
    "    L = pd.DataFrame(L)\n",
    "    M = pd.DataFrame(M)\n",
    "    N = pd.DataFrame(N)\n",
    "    P = pd.DataFrame(P)\n",
    "    Q = pd.DataFrame(Q)\n",
    "    R = pd.DataFrame(R)\n",
    "    S = pd.DataFrame(S)\n",
    "    T = pd.DataFrame(T)\n",
    "    V = pd.DataFrame(V)\n",
    "    W = pd.DataFrame(W)\n",
    "    Y = pd.DataFrame(Y)\n",
    "        \n",
    "    #findf = pd.concat([A,C,D,E,F,G,H,I,K,L,M,N,P,Q,R,S,T,V,W,Y], axis = 1)    \n",
    "    #findf.columns = aalist1'''\n",
    "    return A,C,D,E,F,G,H,I,K,L,M,N,P,Q,R,S,T,V,W,Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "e5536f08",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "([[]], [[]], [[]], [[]], [[]], [[]], [[]], [['ATT', 'ATA', 'ATC']], [['AAG']], [['CTG', 'TTG']], [[]], [[]], [[]], [[]], [['AGG']], [[]], [['ACG']], [['GTG']], [[]], [[]])\n"
     ]
    }
   ],
   "source": [
    "Ltest = formfill_nuc(ube3a_cod,0)\n",
    "print(Ltest)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "cfe66168",
   "metadata": {},
   "outputs": [],
   "source": [
    "formfill = []\n",
    "for i in range (0, len(ube3a_cod)):\n",
    "    idk = formfill_nuc(ube3a_cod,i)\n",
    "    formfill.append(idk)\n",
    "formdf = pd.DataFrame(formfill)\n",
    "formdf.columns = aalist1\n",
    "with pd.ExcelWriter(\"C:/Users/hheath/Documents/aminotest3.xlsx\") as writer:\n",
    "    formdf.to_excel(writer)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc6fe471",
   "metadata": {},
   "source": [
    "PART 2: GROUP AMINO ACIDS, TRACK AMINO ACID GROUP CHANGES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "3c7a89ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "def groupid(codon): #takes a codon, returns the amino acid group\n",
    "    aa = aa_id(codon)\n",
    "    if aa == 'Ala' or aa =='Ile' or aa == 'Leu' or aa == 'Val' or aa == 'Met': \n",
    "        return '1: hydrophobic, small'  \n",
    "    if aa == 'Tyr' or aa == 'Trp' or aa == 'Phe':\n",
    "        return '2: hydrophobic, big'\n",
    "    if aa == 'Lys' or aa == 'Arg' or aa == 'His': \n",
    "        return '3: hydrophobic, positive charge'\n",
    "    if aa == 'Asp' or aa == 'Glu': \n",
    "        return '4:hydrophobic negative charge'\n",
    "    if aa == 'Ser' or aa == 'Thr': \n",
    "        return '5: polar uncharged 1'\n",
    "    if aa == 'Gln' or aa == 'Asn': \n",
    "        return '6: polar uncharged 2'\n",
    "    if aa == 'Gly':\n",
    "        return '7: SPC Gly'\n",
    "    if aa == 'Pro':\n",
    "        return '8: SPC Pro'\n",
    "    if aa == 'Cys': \n",
    "        return '9: SPC Cys'\n",
    "    if aa == 'STOP': \n",
    "        return '10: STOP'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "a43f6031",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "o_cod  = res2['original codon'] #I straight up don't remember what this does. something to do with the groups \n",
    "alt1_1 = res1['nuc1 alt1']\n",
    "alt1_2 = res1['nuc1 alt2']\n",
    "alt1_3 = res1['nuc1 alt3']\n",
    "alt2_1 = res2['nuc2 alt1']\n",
    "alt2_2 = res2['nuc2 alt2']\n",
    "alt2_3 = res2['nuc2 alt3']\n",
    "alt3_1 = res3['nuc3 alt1']\n",
    "alt3_2 = res3['nuc3 alt2']\n",
    "alt3_3 = res3['nuc3 alt3']\n",
    "\n",
    "group_1_1 = []\n",
    "group_1_2 = []\n",
    "group_1_3 = []\n",
    "group_2_1 = []\n",
    "group_2_2 = []\n",
    "group_2_3 = []\n",
    "group_3_1 = []\n",
    "group_3_2 = []\n",
    "group_3_3 = []\n",
    "o_cod_group = []\n",
    "\n",
    "for i in range (0, len(o_cod)): \n",
    "    group = groupid(o_cod[i])\n",
    "    o_cod_group.append(group)\n",
    "df_cod = pd.DataFrame(o_cod_group)\n",
    "\n",
    "for i in range (0, len(alt1_1)): \n",
    "    group = groupid(alt1_1[i])\n",
    "    if group == o_cod_group[i]: \n",
    "        group = 'no_change'\n",
    "    group_1_1.append(group)\n",
    "\n",
    "for i in range (0, len(alt1_2)): \n",
    "    group = groupid(alt1_2[i])\n",
    "    if group == o_cod_group[i]: \n",
    "        group = 'no_change'\n",
    "    group_1_2.append(group)\n",
    "\n",
    "for i in range (0, len(alt1_3)): \n",
    "    group = groupid(alt1_3[i])\n",
    "    if group == o_cod_group[i]: \n",
    "        group = 'no_change'\n",
    "    group_1_3.append(group)\n",
    "\n",
    "for i in range (0, len(alt2_1)): \n",
    "    group = groupid(alt2_1[i])\n",
    "    if group == o_cod_group[i]: \n",
    "        group = 'no_change'\n",
    "    group_2_1.append(group)\n",
    "\n",
    "for i in range (0, len(alt2_2)): \n",
    "    group = groupid(alt2_2[i])\n",
    "    if group == o_cod_group[i]: \n",
    "        group = 'no_change'\n",
    "    group_2_2.append(group)\n",
    "\n",
    "for i in range (0, len(alt2_3)): \n",
    "    group = groupid(alt2_3[i])\n",
    "    if group == o_cod_group[i]: \n",
    "        group = 'no_change'\n",
    "    group_2_3.append(group)\n",
    "    \n",
    "for i in range (0, len(alt3_1)): \n",
    "    group = groupid(alt3_1[i])\n",
    "    if group == o_cod_group[i]: \n",
    "        group = 'no_change'\n",
    "    group_3_1.append(group)\n",
    "\n",
    "for i in range (0, len(alt3_2)): \n",
    "    group = groupid(alt3_2[i])\n",
    "    if group == o_cod_group[i]: \n",
    "        group = 'no_change'\n",
    "    group_3_2.append(group)\n",
    "\n",
    "for i in range (0, len(alt3_3)): \n",
    "    group = groupid(alt3_3[i])\n",
    "    if group == o_cod_group[i]: \n",
    "        group = 'no_change'\n",
    "    group_3_3.append(group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "bb210f35",
   "metadata": {},
   "outputs": [],
   "source": [
    "def flip(l1, l2, l3, l4, l5, l6, l7, l8, l9, ind): #flips axes. used to check rows (not cols) for mutations\n",
    "    flip = []                                      #ex: 0 1 2  flip(col1,col2,col3,0)  \n",
    "    flip.append(l1[ind])                           #    4 5 6         ->               [0,1,2] \n",
    "    flip.append(l2[ind])\n",
    "    flip.append(l3[ind])\n",
    "    flip.append(l4[ind])\n",
    "    flip.append(l5[ind])\n",
    "    flip.append(l6[ind])\n",
    "    flip.append(l7[ind])\n",
    "    flip.append(l8[ind])\n",
    "    flip.append(l9[ind])\n",
    "    return flip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "1ce5e6d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#print(group_1_1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "52b4fa14",
   "metadata": {},
   "outputs": [],
   "source": [
    "def groupchange(i): #checks one row (codon) for group changes. gc = group change count\n",
    "    gc = 0\n",
    "    c1 = flip (group_1_1, group_1_2, group_1_3,group_2_1, group_2_2, group_2_3,group_3_1, group_3_2, group_3_3, i)\n",
    "    for j in range (0, len(c1)): \n",
    "        if c1[j] != 'no_change': \n",
    "            gc += 1\n",
    "    return gc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "f8d8d217",
   "metadata": {},
   "outputs": [],
   "source": [
    "def groupchange2(i):#checks one row(codon) for group changes. gc = UNIQUE group changes in a row\n",
    "    gclst = []\n",
    "    c1 = flip (group_1_1, group_1_2, group_1_3,group_2_1, group_2_2, group_2_3,group_3_1, group_3_2, group_3_3, i) \n",
    "    for j in range (0, len(c1)): \n",
    "        if c1[j] != 'no_change': \n",
    "            gclst.append(c1[j])\n",
    "    gcset = set(gclst)\n",
    "    gc = len(gcset)\n",
    "    return gc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "6638e486",
   "metadata": {},
   "outputs": [],
   "source": [
    "def groupchange3(lst): \n",
    "    changelst = []\n",
    "    for i in range(0, len(lst)): \n",
    "        group = groupid(lst[i])\n",
    "        #if group == groupid(o_cod[i]): \n",
    "        #    changelst.append('NC')\n",
    "        if group != groupid(o_cod[i]): \n",
    "            changelst.append(i)\n",
    "    return changelst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "9b7cc176",
   "metadata": {},
   "outputs": [],
   "source": [
    "def groupchange4 (i): \n",
    "    gclst = []\n",
    "    c1 = flip (group_1_1, group_1_2, group_1_3,group_2_1, group_2_2, group_2_3,group_3_1, group_3_2, group_3_3, i) \n",
    "    for j in range (0, len(c1)): \n",
    "        if c1[j] != 'no_change' and c1[j] != '10: STOP':\n",
    "            gclst.append(c1[j])\n",
    "    gcset = set(gclst)\n",
    "    gc = len(gcset)\n",
    "    return gc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "dcda0675",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'group_1_1' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[35], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m g11 \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame(group_1_1)\n\u001b[0;32m      2\u001b[0m g12 \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame(group_1_2)\n\u001b[0;32m      3\u001b[0m g13 \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame(group_1_3)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'group_1_1' is not defined"
     ]
    }
   ],
   "source": [
    "g11 = pd.DataFrame(group_1_1)\n",
    "g12 = pd.DataFrame(group_1_2)\n",
    "g13 = pd.DataFrame(group_1_3)\n",
    "g21 = pd.DataFrame(group_2_1)\n",
    "g22 = pd.DataFrame(group_2_2)\n",
    "g23 = pd.DataFrame(group_2_3)\n",
    "g31 = pd.DataFrame(group_3_1)\n",
    "g32 = pd.DataFrame(group_3_2)\n",
    "g33 = pd.DataFrame(group_3_3)\n",
    "\n",
    "gc_highlight = pd.concat([g11,g12,g13,g21,g22,g23,g31,g32,g33], axis = 1)\n",
    "\n",
    "gc_highlight.columns = ['g11','g12','g13','g21','g22','g23','g31','g32','g33']\n",
    "#gc_highlight.style.applymap(lambda x: 'background-color : cyan' if x != 'no_change' else '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "55daade1",
   "metadata": {},
   "outputs": [],
   "source": [
    "group_changes  = []\n",
    "group_changes2 = []\n",
    "group_changes4 = []\n",
    "\n",
    "for i in range(0, len(ube3a_cod)): \n",
    "    group_changes.append(groupchange(i))\n",
    "    group_changes2.append(groupchange2(i))\n",
    "    group_changes4.append(groupchange4(i))  \n",
    "        \n",
    "gc_count2 = pd.DataFrame(group_changes2)\n",
    "gc_count  = pd.DataFrame(group_changes)\n",
    "gc_count4 = pd.DataFrame(group_changes4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "264837fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "codnum = pd.DataFrame(codnum)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "729c7529",
   "metadata": {},
   "outputs": [],
   "source": [
    "gc_highlight = pd.concat([codnum, gc_highlight,gc_count,gc_count2], axis = 1)\n",
    "with pd.ExcelWriter(\"C:/Users/hheath/Documents/groups.xlsx\") as writer:\n",
    "    gc_highlight.to_excel(writer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "65524f3d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#d = {'original group': o_cod_group,'1_1': group_1_1, '1_2': group_1_2, '1_3': group_1_3,'2_1': group_2_1, '2_2': group_2_2, '2_3': group_2_3,'3_1': group_3_1, '3_2': group_3_2, '3_3': group_3_3, 'group changes': group_changes}\n",
    "#df = pd.DataFrame(d)\n",
    "#dfgroup = pd.concat([codnumdf, df], axis = 1)\n",
    "#with pd.ExcelWriter(\"C:/Users/hheath/Documents/groups.xlsx\") as writer:\n",
    "#    dfgroup.to_excel(writer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "6381e8c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def gc_cn(group): \n",
    "    cn = []\n",
    "    for i in range (0, len(group)): \n",
    "        if group[i] == 'no_change': \n",
    "            cn.append('')\n",
    "        if group[i] != 'no_change': \n",
    "            cn.append(codnum[i])\n",
    "            cn.append(test[i])\n",
    "    return cn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "639ee6b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "dftest = pd.concat([aminochange,gc_count,gc_count2,gc_count4], axis = 1)\n",
    "dftest.columns = ['codon #', 'original codon', 'AA ID','ignore','nuc1 alt1','aa1 alt1','nuc1 alt2','aa1 alt2','nuc1 alt3','aa1 alt3','mutations','original codon','ignore','nuc2 alt1','aa2 alt1','nuc2 alt2','aa2 alt2','nuc2 alt3','aa2 alt3','mutations','original codon','ignore','nuc3 alt1','aa3 alt1','nuc3 alt2','aa3 alt2','nuc3 alt3','aa3 alt3','mutations','group changes', 'unique group changes', 'unique no stop']\n",
    "with pd.ExcelWriter(\"C:/Users/hheath/Documents/groups.xlsx\") as writer:\n",
    "    dftest.to_excel(writer)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d106961",
   "metadata": {},
   "source": [
    "PART 3: CLINVAR RESULTS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "39c44a10",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar = pd.read_excel(\"C:/Users/hheath/Documents/clinvar_result.xlsx\")\n",
    "CN = list(clinvar['codon#'])\n",
    "gclass = list(clinvar['Germline classification'])\n",
    "classes = set(gclass)\n",
    "lastcol = list(clinvar['use'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "22c04d13",
   "metadata": {},
   "outputs": [],
   "source": [
    "path = []\n",
    "path_lik = []\n",
    "conflict = []\n",
    "lik_ben = []\n",
    "uncertain = []\n",
    "ben_lik = []\n",
    "lik_path = []\n",
    "benign = []\n",
    "\n",
    "for i in range(0, len(gclass)): \n",
    "    if gclass [i] =='Pathogenic': \n",
    "        path.append(CN[i])\n",
    "    if gclass [i] =='Pathogenic/Likely pathogenic':\n",
    "        path_lik.append(CN[i])\n",
    "    if gclass [i] =='Conflicting classifications of pathogenicity':\n",
    "        conflict.append(CN[i])\n",
    "    if gclass [i] =='Likely benign':\n",
    "        lik_ben.append(CN[i])\n",
    "    if gclass [i] =='Uncertain significance':\n",
    "        uncertain.append(CN[i])\n",
    "    if gclass [i] =='Benign/Likely benign':\n",
    "        ben_lik.append(CN[i])\n",
    "    if gclass [i] =='Likely pathogenic':\n",
    "        lik_path.append(CN[i])\n",
    "    if gclass [i] =='Benign':\n",
    "        benign.append(CN[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "4379b42b",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_16284\\3811507357.py\u001b[0m in \u001b[0;36m?\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[0mCVclass\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcodnum\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m     \u001b[1;32mif\u001b[0m \u001b[0mcodnum\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mpath\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      4\u001b[0m         \u001b[0mCVclass\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'Pathogenic'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mcodnum\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mpath_lik\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      6\u001b[0m         \u001b[0mCVclass\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'Pathogenic/Likely Pathogenic'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\AppData\\Local\\anaconda3\\Lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m   1464\u001b[0m     \u001b[1;33m@\u001b[0m\u001b[0mfinal\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1465\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m__nonzero__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[0mNoReturn\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1466\u001b[1;33m         raise ValueError(\n\u001b[0m\u001b[0;32m   1467\u001b[0m             \u001b[1;34mf\"The truth value of a {type(self).__name__} is ambiguous. \"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1468\u001b[0m             \u001b[1;34m\"Use a.empty, a.bool(), a.item(), a.any() or a.all().\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1469\u001b[0m         )\n",
      "\u001b[1;31mValueError\u001b[0m: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all()."
     ]
    }
   ],
   "source": [
    "CVclass = []\n",
    "for i in range(0, len(codnum)): \n",
    "    if codnum[i] in path: \n",
    "        CVclass.append('Pathogenic')\n",
    "    if codnum[i] in path_lik: \n",
    "        CVclass.append('Pathogenic/Likely Pathogenic')\n",
    "    if codnum[i] in conflict: \n",
    "        CVclass.append('Conflicting classifications of pathogenicity')\n",
    "    if codnum[i] in lik_ben: \n",
    "        CVclass.append('Likely Benign')\n",
    "    if codnum[i] in uncertain: \n",
    "        CVclass.append('Uncertain significance')\n",
    "    if codnum[i] in ben_lik: \n",
    "        CVclass.append('Benign/Likely benign')\n",
    "    if codnum[i] in lik_path: \n",
    "        CVclass.append('Likely Pathogenic')\n",
    "    if codnum[i] in benign: \n",
    "        CVclass.append('Benign')\n",
    "    if codnum[i] not in path and codnum [i] not in path_lik and codnum [i] not in conflict and codnum [i] not in lik_ben and codnum [i] not in uncertain and codnum [i] not in ben_lik and codnum [i] not in lik_path and codnum [i] not in benign:\n",
    "            CVclass.append('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d58b926",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5fa0c8cd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "430035ce",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
